src/riscv/context.rs

```diff
@@ -155,6 +196,27 @@ impl TaskContext {
         self.satp = satp;
     }
 
+    /// Gets the TrapFrame for this task by calculating from kernel stack top.
+    ///
+    /// This allows us to modify the TrapFrame's sstatus directly during task switching,
+    /// enabling unified FS state management.
+    #[cfg(feature = "uspace")]
+    unsafe fn get_trapframe_mut(&mut self) -> Option<&mut TrapFrame> {
+        if self.kernel_stack_top != 0 {
```

Comment:
`kernel_stack_top` cannot be used as a basis for determining whether it is a kernel thread or not, and we also need to get the `TrapFrame` of kernel threads through `sscrach`